cmd/osctl/cmd/cluster.go

```diff
@@ -81,12 +81,14 @@ func create() (err error) {
 
 	fmt.Println("generating PKI and tokens")
 
-	ips := []string{"10.5.0.2", "10.5.0.3", "10.5.0.4", "10.5.0.5"}
+	ips := []string{"10.5.0.2", "10.5.0.3", "10.5.0.4"}
 
 	input, err := generate.NewInput(clusterName, ips)
 	if err != nil {
 		return err
 	}
+	input2 := *input
```

Comment:
is there a better way to do this?
In order to get the proper IPs added in for etcd certs and endpoints, I need to massage the `input` parameter. Since this gets returned as a pointer on initial generation, any of the temp data I store there gets overwritten on the next iteration.

Reply:
I think the only way is to pass this by value (and overall, do more pass by value rather pass by pointer). 

I mean `node.Request` to have `Input` field as non-pointer value. I think in Go unless struct is extremely large or contains something which can't be copied (like some OS resource), it's more or less okay to pass by value, as it has less pressure on GC, while copying values is pretty fast in Go.

I can't find a link now, but there was awesome blog post on how Go does copying values in super efficient way via some cool trick.

Reply:
Yeah I think the pass-by-value optimization might help us for edge use cases too. So maybe once we reach beta we can refactor things to use more pass-by-value.